src/diart/models.py

```diff
@@ -7,29 +7,47 @@
 from requests import HTTPError
 
 try:
-    from pyannote.audio import Inference, Model
-    from pyannote.audio.pipelines.speaker_verification import (
-        PretrainedSpeakerEmbedding,
-    )
+    import pyannote.audio.pipelines.utils as pyannote_loader
+    from pyannote.audio.utils.powerset import Powerset
 
     _has_pyannote = True
 except ImportError:
     _has_pyannote = False
 
 
+class PowersetAdapter(nn.Module):
+    def __init__(self, segmentation_model: nn.Module):
+        super().__init__()
+        self.model = segmentation_model
+        specs = self.model.specifications
+        max_speakers_per_frame = specs.powerset_max_classes
+        max_speakers_per_chunk = len(specs.classes)
+        self.powerset = Powerset(max_speakers_per_chunk, max_speakers_per_frame)
+
+    @property
+    def specifications(self):
+        return self.model.specifications
+
+    @property
+    def audio(self):
+        return self.model.audio
+
+    def forward(self, waveform: torch.Tensor) -> torch.Tensor:
+        return self.powerset.to_multilabel(self.model(waveform), soft=False)
+
+
 class PyannoteLoader:
     def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__()
         self.model_info = model_info
         self.hf_token = hf_token
 
-    def __call__(self) -> Callable:
-        try:
-            return Model.from_pretrained(self.model_info, use_auth_token=self.hf_token)
-        except HTTPError:
-            return PretrainedSpeakerEmbedding(
-                self.model_info, use_auth_token=self.hf_token
-            )
+    def __call__(self) -> nn.Module:
+        model = pyannote_loader.get_model(self.model_info, self.hf_token)
+        specs = getattr(model, "specifications", None)
+        if specs is not None and specs.powerset:
+            model = PowersetAdapter(model)
+        return model
```

Comment:
```suggestion
    def __call__(self) -> nn.Module:
        try:
            model = pyannote_loader.get_model(self.model_info, self.hf_token)
            specs = getattr(model, "specifications", None)
            if specs is not None and specs.powerset:
                model = PowersetAdapter(model)
            return model
        except HTTPError:
            return PretrainedSpeakerEmbedding(
                self.model_info, use_auth_token=self.hf_token
            )
```

Something like that?

Reply:
I actually prefer `Model.from_pretrained()` but yes that's the idea